### ISSUE TITLE
Missing records in MeiliSearch index on running scout:import

### ISSUE THREAD

**Issue by koapeadu** [laravel/scout#565]
- Scout Version: 9.3.3
- Scout Driver: MeiliSearch
- Laravel Version: 8.42.1
- PHP Version: 7.4.3
- Database Driver & Version: meilisearch server v0.24

### Description:
Running the `scout:import` is clears the search index before indexing the records. However, given that MeiliSearch does updates in an asynchronous fashion (https://docs.meilisearch.com/learn/advanced/asynchronous_updates.html), the clear index operation might not have been completed by the time the records are indexed. So if you had 5k records, for example, you might end up with just 4.5k records indexed because the clear index operation actually happened after the first 500 records were indexed and hence the were lost.

### Suggestions:

-  Make `scout:import` index only; in which case one has to run `scout:flush` before `scout:import` if the person already had data in the index. The person would be responsible for waiting for MeiliSearch to complete the operation.
-  Alteranatively, modify the import command for meilisearch to await the completion of the clear operation before proceeding with indexing. This can be achieved by polling the status of the resulting _updateId_ till the state changes to `processing` or `failed` (Example here: https://github.com/meilisearch/MeiliSearch/issues/975#issuecomment-698042995)

### Steps To Reproduce:
1. Create a searchable model
2. Seed with 10k+ records but pause indexing during seed
3. Run `php artisan scout:import App\\Models\\ModelClassName`
4. Refresh and check the MeiliSearch dashboard for number of items in index (e.g. http://localhost:7700)
5. Compare with actual row count in database
6. Repeat steps 3-5 a number of times and you would find cases where the number of indexed items is less than actual row count in database.

**Comment by driesvints** [comment]
@mmachatschek @curquiza what do you think the best solution is here?

I also vaguely remember Algolia suffering from the same issue but I'm unsure.

**Comment by curquiza** [comment]
Hello @driesvints and @koapeadu 

Yes Algolia has the same behavior, the indexation is asynchronous.

> the clear index operation might not have been completed by the time the records are indexed

I'm not sure to well understand. What is the clear exactly? The deletion of all the data in the index before the indexation of the new data?
For me, if the clear comes first, it should not create any problem:
1. the "tasks" of document deletion are enqueued into MeiliSearch,
2. then the "tasks" regarding yhe addition of the new documents are enqueued and so are coming after the deletion tasks in the queue
3. The tasks are processed in order: the deletion tasks first and then the additional ones. So you should have your 5000 documents at the end.

Sorry, I know I probably miss something.

However, regarding the two suggestions, I don't think we want to go with the second option for some reasons
-  First, waiting for a task to be done (on Algolia or MeiliSearch side) can take a lot of time, depending on your data, and I'm not sure we want the users to be stuck with the command for a while
- Secondly, it would mean developing a version of the command for MeiliSearch and Algolia separately since they don't handle the tasks the same way in their API.

**Comment by koapeadu** [comment]
Ok I get it. 

I think the problem would probably because I had also configured a redis queue and the Scout queue config was on (Sorry, I should have included that in the Steps to reproduce). My guess is that since there were multiple queue workers the tasks were not necessarily executed in order hence the problem.

As a workaround I switched to the `sync` queue driver and run the import and that worked fine. Then I reverted to the `redis` queue driver.

> I'm not sure to well understand. What is the clear exactly? The deletion of all the data in the index before the indexation of the new data?

That's right.
